### PR TITLE
Rename library entrypoints to Telemetry

### PR DESCRIPTION
**Pull request by anuraaga**
I've been wanting to solve our long standing issue of naming our entrypoints Tracing even though they are more than tracing since we should write more examples for library instrumentation and would be good to have more stable APIs. I think we liked the idea Telemetry when discussing a while back.

### CODE REVIEW

File: instrumentation/graphql-java-12.0/library/src/test/java/io/opentelemetry/instrumentation/graphql/GraphqlTest.java [hunk 1]
```diff
@@ -22,8 +22,8 @@ protected InstrumentationExtension getTesting() {
 
   @Override
   protected void configure(GraphQL.Builder builder) {
-    GraphQLTracing tracing =
-        GraphQLTracing.builder(testing.getOpenTelemetry())
+    GraphQLTelemetry tracing =
+        GraphQLTelemetry.builder(testing.getOpenTelemetry())
             .setCaptureExperimentalSpanAttributes(true)
             .build();
     builder.instrumentation(tracing.newInstrumentation());
```
Comment (mateuszrzeszutek) [inline instrumentation/graphql-java-12.0/library/src/test/java/io/opentelemetry/instrumentation/graphql/GraphqlTest.java:29]:
```suggestion
    GraphQLTelemetry telemetry =
        GraphQLTelemetry.builder(testing.getOpenTelemetry())
            .setCaptureExperimentalSpanAttributes(true)
            .build();
    builder.instrumentation(telemetry.newInstrumentation());
```

File: instrumentation/graphql-java-12.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/graphql/GraphqlSingletons.java [hunk 2]
```diff
@@ -20,8 +20,8 @@
   private static final boolean CAPTURE_EXPERIMENTAL_SPAN_ATTRIBUTES =
       Config.get().getBoolean("otel.instrumentation.graphql.experimental-span-attributes", false);
 
-  private static final GraphQLTracing TRACING =
-      GraphQLTracing.builder(GlobalOpenTelemetry.get())
+  private static final GraphQLTelemetry TRACING =
```
Comment (mateuszrzeszutek) [inline instrumentation/graphql-java-12.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/graphql/GraphqlSingletons.java:23]: `s/TRACING/TELEMETRY/g`

File: instrumentation/armeria-1.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/armeria/v1_3/ArmeriaSingletons.java [hunk 3]
```diff
@@ -19,7 +19,7 @@
   public static final Function<? super HttpService, ? extends HttpService> SERVER_DECORATOR;
 
   static {
-    ArmeriaTracing tracing = ArmeriaTracing.create(GlobalOpenTelemetry.get());
+    ArmeriaTelemetry tracing = ArmeriaTelemetry.create(GlobalOpenTelemetry.get());
 
     CLIENT_DECORATOR = tracing.newClientDecorator();
     SERVER_DECORATOR = tracing.newServiceDecorator();
```
Comment (mateuszrzeszutek) [inline instrumentation/armeria-1.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/armeria/v1_3/ArmeriaSingletons.java:25]:
```suggestion
    ArmeriaTelemetry telemetry = ArmeriaTelemetry.create(GlobalOpenTelemetry.get());

    CLIENT_DECORATOR = telemetry.newClientDecorator();
    SERVER_DECORATOR = telemetry.newServiceDecorator();
```

File: instrumentation/grpc-1.6/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/grpc/v1_6/GrpcSingletons.java [hunk 4]
```diff
@@ -26,8 +26,8 @@
     boolean experimentalSpanAttributes =
         Config.get().getBoolean("otel.instrumentation.grpc.experimental-span-attributes", false);
 
-    GrpcTracing tracing =
-        GrpcTracing.builder(GlobalOpenTelemetry.get())
+    GrpcTelemetry tracing =
```
Comment (mateuszrzeszutek) [inline instrumentation/grpc-1.6/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/grpc/v1_6/GrpcSingletons.java:29]: `s/tracing/telemetry/g`

**Review by mateuszrzeszutek — APPROVED** [review]
Awesome! 🚀

**Review by trask — COMMENTED** [review]
nice 👍

**Pull request merged**